Library/Homebrew/cmd/upgrade.rb

```diff
@@ -104,6 +104,32 @@ def upgrade
         onoe "#{f}: #{e}"
       end
     end
+
+    unless Homebrew.failed?
+      #  Assess existing kegs and reinstall any broken.
+      broken_formulae = []
+
+      CacheStoreDatabase.use(:linkage) do |db|
+        kegs = Formula.installed.collect(&:opt_or_installed_prefix_keg).reject(&:nil?)
+
+        odebug "Assessing keg linkage status" if kegs.size > 1
+
+        kegs.each do |keg|
+          result = LinkageChecker.new(keg, cache_db: db)
+
+          broken_formulae << resolve_formula(keg) if result.broken_library_linkage?
```

Comment:
I get the following error when testing this:

~~~
Error: undefined method `resolve_formula' for Homebrew:Module
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:120:in `block (2 levels) in upgrade'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:117:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:117:in `block in upgrade'
/usr/local/Homebrew/Library/Homebrew/cache_store.rb:16:in `use'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:112:in `upgrade'
/usr/local/Homebrew/Library/Homebrew/brew.rb:87:in `<main>'
~~~

I believe `linkage_checker.rb` has this function, but `cmd/upgrade.rb` does not

Reply:
It should be fixed in the next commit. Thanks for testing!